types/mui-datatables/index.d.ts

```diff
@@ -224,52 +353,166 @@ export interface MUIDataTableIsRowCheck {
+     * This is useful for creating footers for individual columns.
+     * @link https://github.com/gregnb/mui-datatables/blob/master/examples/customize-footer/index.js
+     */
     customTableBodyFooterRender: (options: { data: any[]; selectableRows: SelectableRows; columns: any[] }) => any;
-    customToolbar: () => React.ReactNode;
+    /**
+     * Render a custom Toolbar
+     * @link https://github.com/gregnb/mui-datatables/blob/master/examples/customize-toolbar/CustomToolbar.js
+     */
+    customToolbar: (data: { displayData: DisplayData }) => React.ReactNode;
+    /**
+     * Render a custom selected row ToolBar
+     * @link https://github.com/gregnb/mui-datatables/blob/master/examples/customize-toolbarselect/CustomToolbarSelect.js
+     */
     customToolbarSelect: (
         selectedRows: { data: Array<{ index: number; dataIndex: number }>; lookup: { [key: number]: boolean } },
-        displayData: Array<{ data: any[]; dataIndex: number }>,
+        displayData: DisplayData,
         setSelectedRows: (rows: number[]) => void,
     ) => React.ReactNode;
     /** @deprecated use `selectToolbarPlacement` instead */
     disableToolbarSelect: boolean;
-    draggableColumns: MUIDataTableDraggableColumns;
+    /**
+     * Possible Values:
+     * true       = Button visible and clickable
+     * false      = Button not visible
+     * 'disabled' = Button is visible but not clickable
+     * @defaultValue true
+     */
     download: ToolbarButton;
+    /**
+     * An object of options to change the output of the csv file.
+     * @defaultValue { filename: 'tableDownload.csv', separator: ',' }
+     */
     downloadOptions: Partial<{
         filename: string;
         separator: string;
         filterOptions: Partial<{ useDisplayedColumnsOnly: boolean; useDisplayedRowsOnly: boolean }>;
     }>;
+    /**
+     * An object of options describing how dragging columns should work.
+     * The options are:
+     * enabled: boolean - Indeicated if draggable columns are enabled. Default is false
+     * transitionTime: number - The time in milliseconds it takes for columns to swap postions. Defaults to 300.
+     *
+     * To disable the dragging of a particular column, see the "draggable" option in the columns options.
+     * Dragging a column to a new position updates the columnOrder array and triggers the onColumnOrderChange callback.
+     */
+    draggableColumns: MUIDataTableDraggableColumns;
+    /**
+     * Shadow depth applied to the Paper component.
+     * @defaultValue 4
+     */
     elevation: number;
+    /**
+     * If provided a non-empty string (ex: "."), it will use that value in the column's names to access nested data.
+     * For example, given a enableNestedDataAccess value of "." and a column name of "phone.cell", the column would use the value found in phone:{cell:"555-5555"}
+     * Any amount of nesting will work.
+     * @link https://github.com/gregnb/mui-datatables/blob/master/examples/data-as-objects/index.js
```

Comment:
```suggestion
     * If a non-empty string (ex: `"."`) is provided, it will use that value in the column's names to access nested data.
     *
     * For example, given a value of `"."` for `enableNestedDataAccess` and a column name of `"phone.cell"`, the column would use the value found in phone: `{ cell:"555-5555" }`
     *
     * Any amount of nesting will work.
     * {@link https://github.com/gregnb/mui-datatables/blob/master/examples/data-as-objects/index.js Example}
```